File: Aula_04/c++/04.cpp
#include <iostream>
#include <string>

class Entrada_Cinema{
    public:
    std::string dia = "";
    int hora = 0, minuto = 0;

    double entrada (){
        double valor = 16.00;
        if (hora >= 17 && hora < 24){
            valor = valor * 1.5;
        } 
        if (dia == "Quarta"){
            return 8.00;
        } 
        std::string dias [3] = {"Sexta", "Sabado", "Domingo"};
        for (int i = 0; i < 3; i++){
            if (dias[i] == dia){
                return valor + 4;
            }
        }
        return valor;
    }

    double meia_entrada (){
        double valor = 8.00;
        if (hora >= 17 && hora < 24){
            valor = valor * 1.5;
        } 
        if (dia == "Quarta"){
            return 8.00;
        } 
        std::string dias [3] = {"Sexta", "Sabado", "Domingo"};
        for (int i = 0; i < 3; i++){
            if (dias[i] == dia){
                return valor + 2;
            }
        }
        return valor;
    }
};

int main(){
    Entrada_Cinema ec;

    std::cout << "Digite o dia da sua sessao: ";
    std::cin >> ec.dia;

    std::cout << "Digite o horario da sua sessao: ";
    std::cin >> ec.hora >> ec.minuto;

    std::cout << "Valor esquerda: " << ec.entrada() << std::endl;
    std::cout << "Valor meia-esquerda: " << ec.meia_entrada() << std::endl;

    return 0;
}

File: Aula_04/python/04.py
class Entrada_Cinema:
    def __init__ (self):
        self.dia = ''
        self.hora = 0
        self.minuto = 0

    def entrada (self):
        valor = 16.00
        if self.hora >= 17 and self.hora < 24:
            valor = valor * 1.5
        if self.dia == 'Quarta':
            return 8.00
        if self.dia in ['Sexta', 'Sabado', 'Domingo']:
            return valor + 4
        return valor
        
    def meia_entrada (self):
        valor = 8.00
        if self.hora >= 17 and self.hora < 24:
            valor = valor * 1.5
        if self.dia == 'Quarta':
            return 8.00
        if self.dia in ['Sexta', 'Sabado', 'Domingo']:
            return valor + 2
        return valor
        
ec = Entrada_Cinema()
ec.dia = input("Digite o dia da sua sessão: ").title()
ec.hora, ec.minuto = map(int,input("Digite o horario da sua sessão: ").split(":"))
print(f"Valor entrada: {ec.entrada()}")
print(f"Valor meia-entrada: {ec.meia_entrada()}")